src/main/java/org/truffleruby/language/threadlocal/ThreadAndFrameLocalStorage.java

```diff
@@ -49,13 +49,7 @@ public Object get() {
                 if (otherThreadValues != null) {
                     return otherThreadValues;
                 } else {
-                    otherThreadValues = new ThreadLocal<Object>() {
-                        @Override
-                        protected Object initialValue() {
-                            return ThreadAndFrameLocalStorage.this.initialValue();
-                        }
-
-                    };
+                    otherThreadValues = ThreadLocal.withInitial(() -> ThreadAndFrameLocalStorage.this.initialValue());
```

Comment:
Since we are no longer in an inner class, just `this::initialValue()` should be enough.